src/serac/physics/utilities/quadrature_data.hpp

```diff
@@ -200,49 +273,95 @@ class QuadratureData : public SyncableData {
  */
 template <>
 class QuadratureData<void> {
+  // Doesn't do anything
+public:
+  SERAC_HOST_DEVICE std::nullptr_t operator()(const int, const int) { return nullptr; }
 };
 
 // A dummy global so that lvalue references can be bound to something of type QData<void>
 // FIXME: There's probably a cleaner way to do this, it's technically a non-const global
 // but it's not really mutable because no operations are defined for it
 extern QuadratureData<void> dummy_qdata;
 
+/**
+ * @brief Stores instances of user-defined type for each quadrature point in a mesh
+ * @tparam T The type of the per-qpt data
+ * @pre T must be default-constructible (TODO: Do we want to allow non-default constructible types?)
```

Comment:
> Do we want to allow non-default constructible types?

I don't imagine so, all the examples of user-defined quadrature data I can think of are POD.